TACC/Core-CMS-Custom
# bug/WP-459: Fix dispatch for reaching submitter admin's list registrations page

## Overview
Adjust dispatch() used in SubmittersTable that is currently preventing APCD Submitter admins from reaching their registrations listing page + change user group checks in renewal listing + form
…

## Related

- [WP-459](https://jira.tacc.utexas.edu/browse/WP-459)
…

## Changes
- Now using the base `dispatch()` method from `TemplateView` class (what RegistrationsTable is built on top of, which SubmittersTable is then built from)
- Instead of using `get_user_role` to check user's permissions for renewal listing + form, instead use CMS groups via `has_groups` util function
…

## Testing

1. Go to https://localhost:8000/admin and set your user's Chosen Groups to just `SUBMITTER_ADMIN`
2. Now go to https://localhost:8000/register/list-registration-requests and confirm listing comes up
3. In one of the entries on this listing, click through the 'Actions' dropdown to get to 'Renew Registration' and confirm you are directed to the registration form with the data populated from that record
4. Repeat steps 2 & 3 with the following roles and confirm the following:
    - `APCD_ADMIN`: the same results as `SUBMITTER_ADMIN`
    - `SUBMITTER_USER`: the listing does not show and instead you are redirected to https://localhost:8000/

## UI

…
